matsim/scenariogen/data/preparation.py

```diff
@@ -17,11 +17,13 @@ def prepare_persons(hh, pp, tt, augment=5, max_hh_size=5, core_weekday=False, re
 
     # Augment data using p_weight
     if augment > 1:
-        df = augment_persons(df, augment)
+        # in the cdmx case we do not need to do p_weight * augment = 5 (see method augment_persons)
```

Comment:
prepare_persons should probably be split into multiple function so you can only use these parts that you want in your scenario

Reply:
Would you do this by defining sub-methods / -functions inside of prepare_persons? I can try to do that if that's the way you want to go

Reply:
I will try to do it, as it requires changing the API and design a little bit.